### PR TITLE
2414 dropdown contrast

### PR DESCRIPTION
**Pull request by andrethomas6**
- Added buffer space at bottom of sidebar, allows the sidebar to be manually scrolled better giving a better experience and to prevent the bottom dropdown from being cutoff.
- Updated if the IPTT table height is less than the sidebar height, the sidebar will not auto scroll but just stays at the top.
- Added a resizing event to recalculate the scrolling effect when browser changes size and/or zoom.

### CODE REVIEW

File: js/pages/iptt_report/components/report/reportBody.js [hunk 1]
```diff
@@ -4,20 +4,29 @@ import ReportTableHeader from './tableHeader';
...
             let scrollable = document.documentElement.scrollHeight - window.innerHeight - footer;
+            
+            // Setting the total height of the table. Set time out needed to render the sidebar in correct order when changing programs, otherwise the sidebar's scroll Top will be off.
+            setTimeout(() => {
+                setTableHeight(scrollable);
+            }, 1)
 
             // Setting how much space left a user can scroll to reach the bottom.
-            setBottomScrolling(scrollable - window.scrollY) 
+            setBottomScrolling(parseInt(scrollable - window.scrollY));
```
Comment (sanjuroj) [inline js/pages/iptt_report/components/report/reportBody.js:21]: I'm confused about the user of parseInt here. If one of the inputs is text, this parseInt doesn't help, does it?

File: js/pages/iptt_report/components/sidebar/sidebar.js [hunk 2]
```diff
@@ -1,21 +1,24 @@
 import React, { useEffect, useState } from 'react';
 import IPTTFilterForm from './filterForm';
 
-export default ({bottomScrolling, extend}) => {
+export default ({bottomScrolling, extend, tableHeight}) => {
+    const [startScroll, setStartScroll] = useState(0);
```
Comment (sanjuroj) [inline js/pages/iptt_report/components/sidebar/sidebar.js:5]: The general comments below are useful but additional comments on what some of these specific variables represent would be helpful, particularly bottomScrolling and startScroll.

File: js/pages/iptt_report/components/sidebar/sidebar.js [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
+        sidebarElement.scrollTop = tableHeight <= startScroll ? 0 : startScroll - bottomScrolling;
     }
+
     // Calculate how much of the sidebar is not visible and can be scrolled.
     useEffect(() => {
         let sidebarElement = document.querySelector("#sidebar");
-        let start = sidebarElement.scrollHeight - sidebarElement.clientHeight;
-        setStartScroll(start)
-    }, [])
+        let start = sidebarElement.scrollHeight - sidebarElement.clientHeight - 300;
```
Comment (sanjuroj) [inline js/pages/iptt_report/components/sidebar/sidebar.js:19]: What's the 300 for?

**Review by sanjuroj — CHANGES_REQUESTED** [review]

File: js/pages/iptt_report/components/report/reportBody.js (hunk 1 above)
Comment (andrethomas6) [reply js/pages/iptt_report/components/report/reportBody.js]: You can use parseInt to drop the decimals. I just use it as an alternative to rounding.

File: js/pages/iptt_report/components/sidebar/sidebar.js (hunk 2 above)
Comment (andrethomas6) [reply js/pages/iptt_report/components/sidebar/sidebar.js]: I added a lot more comments, definitions, and an example. Let me know if it makes it easier to understand.

File: js/pages/iptt_report/components/sidebar/sidebar.js (hunk 3 above)
Comment (andrethomas6) [reply js/pages/iptt_report/components/sidebar/sidebar.js]: I increase the height of the last element in the sidebar to 350px, the "#filter-extra" component. This just allows the user to more freely scroll the sidebar and also prevents the bottom dropdowns from being cut off. But I didn't want this extra space to be included when I'm calculating where to start syncing the scroll effect, so I offset the calculation by 300px to account for that added blank space.

File: js/pages/iptt_report/components/report/reportBody.js (hunk 1 above)
Comment (sanjuroj) [reply js/pages/iptt_report/components/report/reportBody.js]: Ah, gotcha, that makes sense.

**Review by sanjuroj — APPROVED** [review]
Super helpful comments!

File: js/pages/iptt_report/components/sidebar/sidebar.js (hunk 2 above)
Comment (sanjuroj) [reply js/pages/iptt_report/components/sidebar/sidebar.js]: The additions were perfect, thanks!

**Pull request merged**